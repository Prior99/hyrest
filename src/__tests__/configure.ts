import { configureController, controller, ControllerMode } from "..";

[
    {
        throwOnError: true,
    },
    {
        errorHandler: (): undefined => undefined,
    },
    {
        baseUrl: "http://example.com/",
    },
    {
        mode: ControllerMode.CLIENT,
    },
    {
        mode: ControllerMode.SERVER,
    },
    {
        throwOnError: true,
        errorHandler: (): undefined => undefined,
        mode: ControllerMode.SERVER,
        baseUrl: "http://example.com/",
    },
].forEach(options => {
    test("`configureController` sets the options as expected", () => {
        @controller()
        class TestController {}

        const testController = new TestController();

        configureController(TestController, options);
        expect(Reflect.getMetadata("api:controller", TestController)).toMatchSnapshot();
    });
});

test("`configureController` throws an error when called with a non-@controller", () => {
    class NotAController {}
    expect(() => configureController(NotAController, {})).toThrowErrorMatchingSnapshot();
});

test("`configureController` with an array of controllers", () => {
    @controller class TestController1 {}
    @controller class TestController2 {}
    @controller class TestController3 {}

    configureController([TestController1, TestController2, TestController3], {
        baseUrl: "http://example.com",
    });
    expect(Reflect.getMetadata("api:controller", TestController1)).toMatchSnapshot();
    expect(Reflect.getMetadata("api:controller", TestController2)).toMatchSnapshot();
    expect(Reflect.getMetadata("api:controller", TestController3)).toMatchSnapshot();
});
