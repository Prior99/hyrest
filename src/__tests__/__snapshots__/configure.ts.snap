// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`configureController\` sets the options as expected 1`] = `
Controller {
  "mode": "client",
  "throwOnError": true,
}
`;

exports[`\`configureController\` sets the options as expected 2`] = `
Controller {
  "errorHandler": [Function],
  "mode": "client",
  "throwOnError": true,
}
`;

exports[`\`configureController\` sets the options as expected 3`] = `
Controller {
  "baseUrl": "http://example.com/",
  "mode": "client",
  "throwOnError": true,
}
`;

exports[`\`configureController\` sets the options as expected 4`] = `
Controller {
  "mode": "client",
  "throwOnError": true,
}
`;

exports[`\`configureController\` sets the options as expected 5`] = `
Controller {
  "mode": "server",
  "throwOnError": true,
}
`;

exports[`\`configureController\` sets the options as expected 6`] = `
Controller {
  "baseUrl": "http://example.com/",
  "errorHandler": [Function],
  "mode": "server",
  "throwOnError": true,
}
`;

exports[`\`configureController\` throws an error when called with a non-@controller 1`] = `"Tried to configure an object which is not a @controller. Check NotAController."`;


